sql/src/parser.rs

```diff
@@ -492,9 +503,42 @@ impl<'a> Parser<'a> {
     }
 }
 
+fn try_build_tskey_constraint(col_defs: &[ColumnDef]) -> Option<TableConstraint> {
```

Comment:
One more thing we may need to take considerations is that it is possible to define multiple tskey constraints and that is not allowed. So we may need to make the result type is: `Result<Option<TableConstraint>>`.